src/ghga_service_commons/utils/utc_dates.py

```diff
@@ -18,65 +18,49 @@
 from __future__ import annotations
 
 from datetime import datetime, timezone
-from typing import Any
+from typing import Annotated
 
-from pydantic import GetCoreSchemaHandler
-from pydantic.type_adapter import TypeAdapter
-from pydantic_core import CoreSchema, core_schema
+from pydantic import AwareDatetime, TypeAdapter
+from pydantic.functional_validators import BeforeValidator
 
-__all__ = ["DateTimeUTC", "UTC", "assert_tz_is_utc", "now_as_utc"]
+__all__ = ["DateTimeUTC", "set_tz_to_utc", "UTC", "assert_tz_is_utc", "now_as_utc"]
 
 UTC = timezone.utc
 
 
-class DateTimeUTC(datetime):
-    """A pydantic type for values that should have an UTC timezone.
-
-    This behaves exactly like the normal datetime type, but requires that the value
-    has a timezone and converts the timezone to UTC if necessary.
-    """
-
-    @classmethod
-    def construct(cls, *args, **kwargs) -> DateTimeUTC:
-        """Construct a datetime with UTC timezone."""
-        if kwargs.get("tzinfo") is None:
-            kwargs["tzinfo"] = UTC
-        return cls(*args, **kwargs)
-
-    @classmethod
-    def __get_pydantic_core_schema__(
-        cls, source_type: Any, handler: GetCoreSchemaHandler
-    ) -> CoreSchema:
-        """Provide the schema with validation function.
-
-        Uses `core_schema.no_info_plain_validator_function` because `cls.validate` is
-        the sole validator for `DateTimeUTC`.
-        """
-        return core_schema.no_info_after_validator_function(
-            cls.validate, handler(datetime)
-        )
-
-    @classmethod
-    def validate(cls, value: Any) -> datetime:
-        """Validate the given value."""
-        datetime_validator = TypeAdapter(datetime)
-        date_value = datetime_validator.validate_python(value)
-        if date_value.tzinfo is None:
-            raise ValueError(f"Date-time value is missing a timezone: {value!r}")
-        if date_value.tzinfo is not UTC:
-            date_value = date_value.astimezone(UTC)
-        return date_value
-
-
 def assert_tz_is_utc() -> None:
     """Verify that the default timezone is set to UTC.
 
-    Raise a Runtimeerror if the default timezone is set differently.
+    Raise a RuntimeError if the default timezone is set differently.
     """
     if datetime.now().astimezone().tzinfo != UTC:
         raise RuntimeError("System must be configured to use UTC.")
 
 
+def set_tz_to_utc(date: datetime) -> datetime:
+    """Force UTC timezone for date."""
```

Comment:
```suggestion
def convert_tz_to_utc(date: datetime) -> datetime:
    """Convert the timezone of the given datetime object to UTC."""
```

Just to make clear that we do not forcefully set or replace the timezone, but convert a possible existing timezone (or the system timezone for a naive object) to UTC.

Reply:
Agreed, that is better. It's in [e26ddfb](https://github.com/ghga-de/ghga-service-commons/pull/26/commits/e26ddfbacfdc24f2774fb542bfd317683c581c95)